platform/core/src/services/ToolBarService/ToolBarService.ts

```diff
@@ -52,19 +52,32 @@ export default class ToolBarService {
     this.buttons = {};
   }
 
+  onModeEnter() {
+    this.reset();
+  }
```

Comment:
The same as the other comment, this reset can be called on a modeEnter in the mode configuration factory. Having a `onModeEnter` on a Service is super weird to me

Reply:
The problem is that it ties the Mode.tsx to the service definitions, so in order to add a new "reset" type service, you have to modify the Mode.tsx in core.  Suppose CornerstoneViewportService suddenly needs a reset - if it is in the service, then you just call it, and everything is good.  However, if it is in the Mode.tsx definition, then you need to update the @ohif/ui core code.  Again, that is what i mean by breaking encapsulation.

Reply:
Added the onModeEnter and onModeExit as requested.